presto-common/src/main/java/com/facebook/presto/common/block/Block.java

```diff
@@ -229,11 +230,21 @@ default long getApproximateRegionLogicalSizeInBytes(int position, int length)
     }
 
     /**
-     * Returns the size of of all positions marked true in the positions array.
+     * Returns the number of bytes (in terms of {@link Block#getSizeInBytes()}) required per position
+     * that this block contains, assuming that the number of bytes required is a known static quantity
+     * and not dependent on any particular specific position. This allows for some complex block wrappings
+     * to potentially avoid having to call {@link Block#getPositionsSizeInBytes(boolean[], int)}  which
+     * would require computing the specific positions selected
+     * @return The size in bytes, per position, if this block type does not require specific position information to compute its size
+     */
+    OptionalInt fixedSizeInBytesPerPosition();
```

Comment:
wonder if we can have more unit test coverage for this method for various block implementations? Especially this could be broken if someone adds a new data structure in some block. A quick way could be using `getRetainedSizeInBytes() / getPositionCount()` to compare with `fixedSizeInBytesPerPosition()` to see if they are close enough when there are lot of entries in a block.

Reply:
Good call out about the tests, I've added a new assertion to `AbstractTestBlock#assertBlockSize` to ensure that all blocks reporting a fixed size in bytes per position are self-consistent between `Block#getSizeInBytes() == Block#getPositionCount() * Block#fixedSizeInBytesPerPosition()`.

I'm not sure about how to correlate this to `Block#getRetainedSizeInBytes()` though since block views (eg: `Block#getRegion(position, length)`) retain more memory than "sizeInBytes" reports, by design.